### PR TITLE
feat: add salvage functionality for security-sensitive values in GlobalContext

### PR DESCRIPTION
**Pull request by Patrick-Erichsen**
Closes CON-2978

## Summary

This PR addresses the security concern raised in #6683 about preserving allowAnonymousTelemetry settings when globalContext.json becomes corrupted.

## Background

This implements a salvage function similar to the existing `salvageSharedConfig` function that was previously used in the GlobalContext class. You can see the original salvage pattern in the git history here: https://github.com/continuedev/continue/commit/3b335b6f8c7c3cc53cf7219cbfc1cef1c4b7ddb5#diff-640bf7a56c89e4e8cdb5efbf5a20a5c5d4c7e5b4d1e8a9a2a8e5e0e2e8a3a7c7 where the `getSharedConfig()` method used `salvageSharedConfig(sharedConfig)` to preserve security-sensitive values when sharedConfig was corrupted.

## Changes

- **Added `salvageGlobalContext` method**: Follows the same pattern as the existing `salvageSharedConfig` function, attempting to extract security-sensitive values from corrupted JSON files before deletion
- **Enhanced error handling**: Updated the `update` method to salvage values before recreating corrupted files  
- **Comprehensive test suite**: Added 6 new test cases covering various salvage scenarios and edge cases

## Security Benefits

- Preserves allowAnonymousTelemetry user privacy preferences even when file corruption occurs
- Prevents loss of security-sensitive configuration data
- Maintains user consent choices for telemetry collection

## Testing

All 18 tests pass, including:
- ✅ Salvaging valid allowAnonymousTelemetry values (both true and false)
- ✅ Handling cases where no sharedConfig exists  
- ✅ Graceful handling of invalid or unparseable content
- ✅ Preserving salvaged values across different update operations

## Backward Compatibility

The salvage functionality is transparent to existing code - if salvage fails, the system continues to work normally by falling back to the original behavior of creating a fresh file.

Addresses the security concern from PR #6683.

### CODE REVIEW

**Comment by Patrick-Erichsen** [comment]
Not sure why the docs changes are showing up here, but those changes are already in `main` so they can be ignored.

**Review by RomneyDa — CHANGES_REQUESTED** [review]
The functionality appears duplicate of the existing `salvageSharedConfig` function. Let's keep one of them and remove the other

**Comment by Patrick-Erichsen** [comment]
Ah yeah good callout 😅  addressed

**Comment by RomneyDa** [comment]
@Patrick-Erichsen prettier check

File: docs/customize/model-providers/top-level/azure.mdx [hunk 1]
```diff
@@ -100,6 +98,7 @@ We recommend configuring **text-embedding-3-large** as your embeddings model.
         env:
           deployment: <YOUR_DEPLOYMENT_NAME>
           apiType: azure
+          apiVersion: 2023-07-01-preview
```
Comment (RomneyDa) [inline docs/customize/model-providers/top-level/azure.mdx:101]: @Patrick-Erichsen these changes seem unrelated was it branched off of something that hasn't been merged?
